### PR TITLE
BUG: DatetimeArray.unit when constructed from a non-nano ndarray

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file if fixing a bug or adding a new feature.

fixes the following inconsistency:

```
from pandas.core.arrays.datetimes import DatetimeArray
import numpy as np

data = np.array([1], dtype="datetime64[s]")
dta = DatetimeArray(data)

print(dta.unit)     # ns
print(dta[0].unit)  # s
```

### CODE REVIEW

File: doc/source/whatsnew/v2.1.0.rst [hunk 1]
```diff
@@ -219,6 +219,7 @@ Datetimelike
 - Bug in :meth:`Timestamp.round` with values close to the implementation bounds returning incorrect results instead of raising ``OutOfBoundsDatetime`` (:issue:`51494`)
 - :meth:`DatetimeIndex.map` with ``na_action="ignore"`` now works as expected. (:issue:`51644`)
 - Bug in :meth:`arrays.DatetimeArray.map` and :meth:`DatetimeIndex.map`, where the supplied callable operated array-wise instead of element-wise (:issue:`51977`)
+- Bug in :class:`arrays.DatetimeArray` constructor returning an incorrect unit when passed a non-nanosecond numpy datetime array (:issue:`52555`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:222]: I think it would be okay to backport this to 2.0.1

Comment (lukemanley) [reply doc/source/whatsnew/v2.1.0.rst]: moved to 2.0.1 - thanks

**Review by mroeschke — APPROVED** [review]
Merge conflict otherwise LGTM

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lukemanley

**Pull request merged**